api/v1beta1/condition_consts.go

```diff
@@ -276,6 +276,10 @@ const (
 	// TopologyReconciledHookBlockingReason (Severity=Info) documents reconciliation of a Cluster topology
 	// not yet completed because at least one of the lifecycle hooks is blocking.
 	TopologyReconciledHookBlockingReason = "LifecycleHookBlocking"
+
+	// TopologyReconciledClusterClassOutdatedReason (Severity=Info) documents reconciliation of a Cluster topology not
+	// yet completed because the ClusterClass `status.observedGeneration` is not the same as its `metadata.generation`.
```

Comment:
```suggestion
	// TopologyReconciledClusterClassOutdatedReason (Severity=Info) documents reconciliation of a Cluster topology not
	// yet completed because the ClusterClass hasn't been successfully reconciled (i.e. `status.observedGeneration`
	// is not the same as its `metadata.generation`.
```
nit: Maybe just slightly easier for users as it's slightly more high-level while still containing the details

Reply:
Q. should we rephrase this from a user PoV, like e.g. waiting for CC to reconcile? 
(trying to avoid questions like: what I have to do to fix the "outdated" CC)

Also, "LifecycleHookBlocking" seems not in line with name/description

Reply:
> Q. should we rephrase this from a user PoV, like e.g. waiting for CC to reconcile?
> (trying to avoid questions like: what I have to do to fix the "outdated" CC)

That's roughly what I wanted to suggest, yup. 

> Also, "LifecycleHookBlocking" seems not in line with name/description
Good catch, absolutely. I saw it initially but somehow forgot to comment on it.

Reply:
What about "ClusterClassNotReconciled" I've tried to put additional info in the message that goes along with the condition too.

Reply:
Sounds good to me

Reply:
But would maybe be still good to change the godoc a bit to make it more user-friendly (as suggested above)